File: covid-api/src/main/java/com/tacs/grupo1/covid/api/dto/statistics/CountryListPlot.java
package com.tacs.grupo1.covid.api.dto.statistics;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.tacs.grupo1.covid.api.dto.CountryList;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
public class CountryListPlot {
    private Long id;
    private String name;
    @JsonProperty("country_plot_list")
    private List<CountryPlot> countryPlotList;
}

File: covid-frontend/src/models/CountryListPlot.ts
import CountryPlot from './CountryPlot'

export default class CountryListPlot{
    public id: number;
    public name: string;
    public country_plot_list: Array<CountryPlot>;

    constructor(countryListPlot?: any){
        if(countryListPlot){
            this.id = countryListPlot.id;
            this.name = countryListPlot.name;
            this.country_plot_list = countryListPlot.country_plot_list;
        }else{
            this.id = 0;
            this.name = "";
            this.country_plot_list = new Array<CountryPlot>();
        }
    }
}